core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala

```diff
@@ -50,7 +51,17 @@ class NettyBlockRpcServer(
       client: TransportClient,
       rpcMessage: ByteBuffer,
       responseContext: RpcResponseCallback): Unit = {
-    val message = BlockTransferMessage.Decoder.fromByteBuffer(rpcMessage)
+    val message = try {
+      BlockTransferMessage.Decoder.fromByteBuffer(rpcMessage)
+    } catch {
+      case _: IndexOutOfBoundsException =>
+        // Netty throws IndexOutOfBoundsException for corrupted buffers. In this case, we ignore
+        // the entire message with warnings because we cannot trust any contents.
+        logWarning(s"Ignored a corrupted RPC message (capacity: ${rpcMessage.capacity()}) " +
+          s"from ${client.getSocketAddress}. Please use `spark.authenticate.*` configurations " +
+          "in case of security incidents.")
```

Comment:
If the incoming message is not valid, we can end up with `IllegalArgumentException`, `IndexOutOfBoundsException`, `IllegalArgumentException`, `NegativeArraySizeException`, etc.

Any particular reason to suspect `IndexOutOfBoundsException` ?

Reply:
Thank you for review. Yes, this code path has also more exception types like `RuntimeException` (due to `IOException`) and `UnsupportedOperationException`. We can add some of them more.

However, currently, this PR focused on random attacks which causes failures mostly at the first field, `appId` decoding which are observed during testing.
```
String appId = Encoders.Strings.decode(buf);
```

Reply:
Might be a good idea to expand to atleast include the common patterns we can observe. Top of my head, the following are possible:
* `IllegalArgumentException` - when the `type` is unknown/invalid when decoding.
* `NegativeArraySizeException` - if the size read is invalid (negative for example).
* `IndexOutOfBoundsException` - we are already handling it in this PR.

Rest can be enhanced in future works, thoughts ?

Reply:
Got i. Thank you for your advice, @mridulm .